src/python/turicreate/toolkits/object_detector/object_detector.py

```diff
@@ -269,14 +269,16 @@ def create(dataset, annotations=None, feature=None, model='darknet-yolo',
         tf_config = {
             'grid_height': params['grid_shape'][0],
             'grid_width': params['grid_shape'][1],
-            'max_iterations': max_iterations,
             'mlmodel_path' : params['mlmodel_path'],
             'classes' : classes,
             'compute_final_metrics' : False
         }
         if batch_size > 0:
             tf_config['batch_size'] = batch_size
 
+        if max_iterations > 0:
```

Comment:
fix bug for OD which is the reason of the failure of test_create_single_image

Reply:
what do you mean?

Reply:
the default value on the python side for `max_iteration` is `0`, which causes an assert error in c++ size which assert `max_iteration` > 0.
So the fix here didn't pass the argument if `max_iteration == 0`, so in the c++ side,
an `estimate_max_iteration()` function is called to produce `max_iterations` automatically.

Reply:
good. You should leave a comment max_iteration is selected automatically if not provided.

Reply:
done thanks!